airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.json

```diff
@@ -15,15 +15,82 @@
       },
       "credentials": {
         "type": "object",
-        "title": "api key",
-        "required": ["api_key"],
-        "properties": {
-          "api_key": {
-            "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
-            "type": "string",
-            "airbyte_secret": true
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "api key",
+            "required": ["api_key", "credentials_title"],
+            "properties": {
+              "credentials_title": {
+                "type": "string",
+                "const": "API Key Credentials",
+                "enum": ["API Key Credentials"],
+                "default": "API Key Credentials",
+                "order": 0
+              },
+              "api_key": {
+                "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
+                "type": "string",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "oauth",
+            "required": [
+              "redirect_uri",
+              "client_id",
+              "client_secret",
+              "refresh_token",
+              "access_token",
+              "credentials_title"
+            ],
+            "properties": {
+              "credentials_title": {
+                "type": "string",
+                "const": "OAuth Credentials",
+                "enum": ["OAuth Credentials"],
+                "default": "OAuth Credentials",
+                "order": 0
+              },
+              "redirect_uri": {
```

Comment:
you dont need redirect_uti, access_token and token_expires parameters for oauth flow.  You will obtain refresh token so access_token could be generated by oauth refresh_token grant call. Also it said that access [lives for 6 hours](https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-and-refresh-tokens) so you cn adjust Oauth authenticator.

Reply:
But these params are not required. Both `access_token` and `token_expires` may be set to `null`. There is just an option to set the `access_token` and `token_expires` values, so we would avoid additional requests on each connector read.

Reply:
It said those field are required:
![image](https://user-images.githubusercontent.com/437607/135107512-bebea4e7-0a9b-45c1-a21a-4c0331f4c3f4.png)
You are not going to need access token / redirect_uri for updating refresh token, just remove those parameters.